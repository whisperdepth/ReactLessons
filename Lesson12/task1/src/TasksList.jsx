/* eslint-disable class-methods-use-this */
import React from 'react';
import Task from './Task.jsx'
import CreateTaskInput from './CreateTaskInput.jsx';

class TasksList extends React.Component {
  state = {
    tasks: [
      { text: 'Buy milk', done: false, id: 1 },
      { text: 'Pick up Tom from airport', done: false, id: 2 },
      { text: 'Visit party', done: false, id: 3 },
      { text: 'Visit doctor', done: true, id: 4 },
      { text: 'Buy meat', done: true, id: 5 },
    ],
  };

  handleCreate = (text) => {
     this.setState({tasks: this.state.tasks.push({text, done: false, id: Math.random()})});
     console.log(this.state.tasks);
  }
   


  render() {
    return (
      <div className='todo-list'>
        <CreateTaskInput onCreate={this.handleCreate}/>
        <ul className='list'>
          {this.state.tasks.map((task) => (
            <Task key={task.id} {...task} />
          ))}
        </ul>
      </div>
    );
  }
    
}

export default TasksList;