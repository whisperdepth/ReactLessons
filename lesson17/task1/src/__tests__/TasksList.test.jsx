import React from "react";
import { shallow } from "enzyme";
import TasksList from "../TasksList";
import { fetchTasksList } from "../tasksGateway";

jest.mock("../tasksGateWay", () => {
  return {
    fetchTasksList: jest.fn(() => Promise.resolve([])),
  };
});

describe("<TasksList />", () => {
  it("should display sorted tasks", () => {
    const wrapper = shallow(<TasksList />);
    wrapper.setState({
      tasks: [
        { id: "1", text: "task1", done: true },
        { id: "2", text: "task2", done: false },
        { id: "3", text: "task3", done: true },
      ],
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("should request tasks list", () => {
    shallow(<TasksList />);

    expect(fetchTasksList).toBeCalled();
  });

});
