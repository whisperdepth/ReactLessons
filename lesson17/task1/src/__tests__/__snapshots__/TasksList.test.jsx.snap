// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksList /> should display sorted tasks 1`] = `
<div
  className="todo-list"
>
  <CreateTaskInput
    onCreate={[Function]}
  />
  <ul
    className="list"
  >
    <Task
      done={false}
      id="2"
      key="2"
      onChange={[Function]}
      onDelete={[Function]}
      text="task2"
    />
    <Task
      done={true}
      id="1"
      key="1"
      onChange={[Function]}
      onDelete={[Function]}
      text="task1"
    />
    <Task
      done={true}
      id="3"
      key="3"
      onChange={[Function]}
      onDelete={[Function]}
      text="task3"
    />
  </ul>
</div>
`;
